extensions/mssql/src/objectManagement/ui/serverPropertiesDialog.ts

```diff
@@ -206,4 +220,108 @@ export class ServerPropertiesDialog extends ObjectManagementDialogBase<Server, S
 		}
 		return errors;
 	}
+
+	private initializeProcessorsSection(): void {
+		const isEnabled = this.engineEdition !== azdata.DatabaseEngineEdition.SqlManagedInstance;
+		let nodes: NumaNode[] = this.objectInfo.numaNodes;
+		let nodeTableList: azdata.TableComponent[] = [];
+		let tableGroups: azdata.GroupContainer[] = [];
+		for (let node of nodes) {
+			let table = this.createProcessorTable(node);
+			nodeTableList.push(table);
+			tableGroups.push(this.createGroup(localizedConstants.serverNumaNodeLabel(node.numaNodeId), [table], true));
+		}
+		this.autoSetProcessorAffinityMaskForAllCheckbox = this.createCheckbox(localizedConstants.autoSetProcessorAffinityMaskForAllText, async (newValue) => {
+			this.objectInfo.autoProcessorAffinityMaskForAll = newValue;
+			for (let table of nodeTableList) {
+				let newData = table.data;
+				for (let i = 0; i < table.data.length; i++) {
+					if (newValue) {
+						// if affinity mask for all is checked, then uncheck the individual processors
+						newData[i][1] = false;
+					}
+				}
+				await this.setTableData(table, newData);
+			}
+		}, this.objectInfo.autoProcessorAffinityMaskForAll, isEnabled);
+
+		this.autoSetProcessorIOAffinityMaskForAllCheckbox = this.createCheckbox(localizedConstants.autoSetProcessorAffinityIOMaskForAllText, async (newValue) => {
+			this.objectInfo.autoProcessorAffinityIOMaskForAll = newValue;
+			for (let table of nodeTableList) {
+				let newData = table.data;
+				for (let i = 0; i < table.data.length; i++) {
+					if (newValue) {
+						// if IO affinity mask for all is checked, then uncheck the individual processors
+						newData[i][2] = false;
+					}
+				}
+				await this.setTableData(table, newData);
+			}
+		}, this.objectInfo.autoProcessorAffinityIOMaskForAll, isEnabled);
+
+		this.processorsSection = this.createGroup('', [
+			this.autoSetProcessorAffinityMaskForAllCheckbox,
+			this.autoSetProcessorIOAffinityMaskForAllCheckbox,
+		], false);
+
+		this.processorsSection.addItems(tableGroups);
+		this.processorsTab = this.createTab(this.processorsTabId, localizedConstants.ProcessorsText, this.processorsSection);
+	}
+
+	private createProcessorTable(numaNode: NumaNode): azdata.TableComponent {
+		const cssClass = 'no-borders';
+		let tableData = numaNode.processors.map(row => [localizedConstants.serverCPULabel(row.processorId), row.affinity, row.ioAffinity]);
+		let processorTable = this.createTable(localizedConstants.processorLabel,
+			[
+				<azdata.TableColumn>{
+					name: localizedConstants.processorColumnText,
+					value: localizedConstants.processorColumnText,
+					type: azdata.ColumnType.text,
+					cssClass: cssClass,
+					headerCssClass: cssClass,
+				},
+				<azdata.TableColumn>{
+					name: localizedConstants.processorAffinityColumnText,
+					value: localizedConstants.processorAffinityColumnText,
+					type: azdata.ColumnType.checkBox,
+					width: DefaultInputWidth / 2,
```

Comment:
Why / 2? Should this actually be tied to the input width value or be its own separate thing?